training/deepspeech_training/util/signal_augmentations.py

```diff
@@ -0,0 +1,200 @@
+
+import os
+import re
+import math
+import random
+import numpy as np
+
+from multiprocessing import Queue, Process
+from .audio import gain_db_to_ratio, max_dbfs, normalize_audio, AUDIO_TYPE_NP, AUDIO_TYPE_PCM, AUDIO_TYPE_OPUS
+from .helpers import int_range, float_range, pick_value_from_range, MEGABYTE
+
+SPEC_PARSER = re.compile(r'^([a-z]+)(\[(.*)\])?$')
+BUFFER_SIZE = 1 * MEGABYTE
+
+
+def _enqueue_overlay_samples(sample_source, queue, buffering=BUFFER_SIZE):
+    # preventing cyclic import problems
+    from .sample_collections import samples_from_source  # pylint: disable=import-outside-toplevel
+    samples = samples_from_source(sample_source, buffering=buffering, labeled=False)
+    while True:
+        for sample in samples:
+            queue.put(sample)
```

Comment:
This is only moving (and buffering) the CSV/SDB `__getitem__` cost to a separate process, right? Shouldn't the `change_audio_type` operation also be done here?

Reply:
Its main purpose is to be a central distribution point for overlay samples running in just one process. This approach ensures that overlay samples are not used twice. It also "just" loads the (raw and still compressed) data and distributes it to the actual overlay workers. They are then doing the decompression, potential conversion and overlaying in parallel.

Reply:
Perfect, thanks for the explanation. Could you put this into a code comment?